backend/qualibrate_app/config/resolvers.py

```diff
@@ -30,3 +30,16 @@ def get_settings(
     config_path: Annotated[Path, Depends(get_config_path)],
 ) -> QualibrateConfig:
     return get_qualibrate_config(config_path)
+
+
+@lru_cache
+def get_quam_state_path(
+    settings: Annotated[QualibrateConfig, Depends(get_settings)],
+) -> Optional[Path]:
+    root = settings.__class__._root
+    if root is None:
+        return None
+    quam_state_path = root._raw_dict.get("quam", {}).get("state_path")
+    if quam_state_path is None:
+        return None
+    return Path(quam_state_path)
```

Comment:
if `quam.state_path` cannot be found, it should also check whether `active_machine.state_path` exists. If it does, it should use that and raise this deprecation warning "The config entry "active_machine.path" has been deprecated in favor of "quam.state_path". Please update the qualibrate config (~/.qualibrate/config.toml) accordingly"